src/util/positioning.ts

```diff
@@ -154,3 +154,9 @@ export function positionElements(
   targetElement.style.top = `${pos.top}px`;
   targetElement.style.left = `${pos.left}px`;
 }
+
+type PlacementSimpleType = 'top' | 'bottom' | 'left' | 'right';
+type PlacementExtendedType = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' | 'left-top' | 'left-bottom' |
+    'right-top' | 'right-bottom';
+
+export type PlacementType = PlacementSimpleType | PlacementExtendedType;
```

Comment:
Did you check how this type renders on our documentation site? Keep in mind that we are generating documentation automatically by walking Typescript AST. We need to make sure that users see a full list of options (`top`, `top-left` etc.) and not just `PlacementType` (or, alternatively, we would have to include documentation for the `PlacementType`)